gprofiler/profilers/node.py

```diff
@@ -123,53 +126,52 @@ def _get_node_pid(sock: WebSocket) -> int:
     except json.JSONDecodeError:
         raise NodeDebuggerUnexpectedResponse(message)
     try:
-        return cast(int, message["result"]["result"]["value"])
+        return message["result"]["result"]["value"]
     except KeyError:
         raise NodeDebuggerUnexpectedResponse(message)
 
 
-def _load_dso(sock: WebSocket, module_path: str) -> None:
+def _change_dso_state(sock: WebSocket, module_path: str, action: str) -> None:
+    assert action in ("start", "stop"), "_change_dso_state supports only start and stop actions"
     cdp_request = {
         "id": 1,
         "method": "Runtime.evaluate",
         "params": {
-            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").start()',
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").{action}()',
             "replMode": True,
         },
     }
     _send_socket_request(sock, cdp_request)
 
 
-def _stop_dso(sock: WebSocket, module_path: str) -> None:
-    cdp_request = {
-        "id": 1,
-        "method": "Runtime.evaluate",
-        "params": {
-            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").stop()',
-            "replMode": True,
-        },
-    }
-    _send_socket_request(sock, cdp_request)
-
-
-def validate_ns(nspid: int, expected_ns_link_name: str) -> None:
+def _validate_ns(nspid: int, expected_ns_link_name: str) -> None:
     actual_nspid_name = os.readlink(os.path.join("/proc", str(nspid), "ns", "pid"))
     assert (
         actual_nspid_name == expected_ns_link_name
     ), f"Wrong namespace, expected {expected_ns_link_name}, got {actual_nspid_name}"
 
 
-def validate_pid(expected_pid: int, sock: WebSocket) -> None:
-    actual_pid = _get_node_pid(sock)
-    assert expected_pid == actual_pid, f"Wrong pid, expected  {expected_pid}, actual {actual_pid}"
+def _validate_ns_node(sock: WebSocket, nspid: str, expected_ns_link_name: str) -> None:
+    command = f'const fs = process.mainModule.require("fs"); fs.readlinkSync("/proc/{nspid}/ns/pid")'
+    actual_ns_link_name = cast(str,_execute_js_command(sock, command))
+    logger.debug(f"Found link by node: {actual_ns_link_name}")
+    assert (
+        actual_ns_link_name == expected_ns_link_name
+    ), f"Wrong namespace, expected {expected_ns_link_name}, got {actual_ns_link_name}"
+
+
+def _validate_pid(expected_pid: int, sock: WebSocket) -> None:
+    actual_pid = cast(int, _execute_js_command(sock, "process.pid"))
+    assert expected_pid == actual_pid, f"Wrong pid, expected {expected_pid}, actual {actual_pid}"
 
 
 def create_debugger_socket(nspid: int, ns_link_name: str) -> WebSocket:
-    validate_ns(nspid, ns_link_name)
+    _validate_ns(nspid, ns_link_name)
```

Comment:
I don't think this check is required at all - this `create_debugger_socket` function is called by us in the PID namespace of the target process. There's no way that the PID namespace won't match here. Only in `_validate_ns_node` there's a chance for failure, because we could be communicating with a Node process that's in the same net NS as us, but different PID NS.

Reply:
Let's remove it